src/Uno.UWP/Devices/Sensors/Accelerometer.Android.cs

```diff
@@ -82,31 +97,37 @@ private void StopShaken()
 				_shakeListener.Dispose();
 				_shakeListener = null;
 			}
+
+			_accelerometer?.Dispose();
+			_accelerometer = null;
 		}
 
 		class ReadingChangedListener : Java.Lang.Object, ISensorEventListener
 		{
 			private readonly Accelerometer _accelerometer;
-			private long _lastTimestamp;
 
 			public ReadingChangedListener(Accelerometer accelerometer)
 			{
 				_accelerometer = accelerometer;
 			}
 
-			public void OnAccuracyChanged(Sensor sensor, [GeneratedEnum] SensorStatus accuracy)
+			public void OnAccuracyChanged(Sensor? sensor, [GeneratedEnum] SensorStatus accuracy)
 			{
 			}
 
-			public void OnSensorChanged(SensorEvent e)
+			public void OnSensorChanged(SensorEvent? e)
 			{
+				if (e is null)
+				{
+					return;
+				}
+
 				var reading = new AccelerometerReading(
-					e.Values[0] / SensorManager.GravityEarth * -1,
-					e.Values[1] / SensorManager.GravityEarth * -1,
-					e.Values[2] / SensorManager.GravityEarth * -1,
+					e.Values![0] / SensorManager.GravityEarth * -1,
+					e.Values![1] / SensorManager.GravityEarth * -1,
+					e.Values![2] / SensorManager.GravityEarth * -1,
```

Comment:
If would be best to avoid this change, let's validate explicitly for null `e.Values`